packages/blockchain/src/blockchain.ts

```diff
@@ -605,8 +607,25 @@ export class Blockchain implements BlockchainInterface {
     await this.validateHeader(block.header)
     await this._validateUncleHeaders(block)
     await block.validateData(false)
+    await this._validateBlobTransactions(block)
   }
 
+  private async _validateBlobTransactions(block: Block) {
```

Comment:
@holgerd77 @jochem-brouwer Curious about this blob transaction validation step for your thoughts.  The spec has this:
```
def validate_block(block: Block) -> None:
    ...

    for tx in block.transactions:
        ...

        # the signer must be able to afford the transaction
        assert signer(tx).balance >= tx.message.gas * tx.message.max_fee_per_gas + get_total_data_gas(tx) * tx.message.max_fee_per_data_gas

        # ensure that the user was willing to at least pay the current data gasprice
        assert tx.message.max_fee_per_data_gas >= get_data_gasprice(parent(block).header)
```

I originally put part of the check in `blockchain` where I thought it made sense since we can validate the transaction's `maxFeePerDataGas` versus the calculated `dataGasFee` from the current headblock but then I realized we need this same check in the `vm` inside `runTx` (how we have similar checks for EIP1559).  Does it make sense to keep this check here as well or just do all this checking inside the `vm`?

Reply:
Hmm, interesting.

I wonder if we have somewhat structurally "forgotten" the 1559 checks in blockchain and just didn't stumble upon it yet because we normally run in an execution context (we run blocks through the VM before adding to blockchain) and therefore didn't notice? 🤔

So I would in a first iteration say we might want to have these checks (so, both the new ones from 4844 and the old one from 1159 in both VM and blockchain)? But just a first guess, would need some confirmation.

Regarding the blob txs, *do* these actually run in any form through the `VM` with `runTx()`? I thought this is just - well - a blob 😋 and therefore just passed on to the blockchain (what would be a strong argument that we would *in these case* definitely need in blockchain?

Reply:
We need access to the `stateManager` because one of the checks is verifying the sender has sufficient balance to execute the tx and we don't have that in blockchain.  So...I either need to figure out how to run that check in `blockchain` (which I don't think is feasible), keep it in `vm`, or it has to go somewhere in `client` which seems like an odd place to put it.

Reply:
>Regarding the blob txs, do these actually run in any form through the VM with runTx()? I thought this is just - well - a blob yum and therefore just passed on to the blockchain (what would be a strong argument that we would in these case definitely need in blockchain?

From my understanding, the blobs are only used when validating the network wrapper since we feed them into the `verifyProof` call to the kzg library.